tsconfig.base.esm.json

```diff
@@ -1,7 +1,7 @@
 {
-  "extends": "./tsconfig.base.es5.json",
+  "extends": "./tsconfig.base.json",
   "compilerOptions": {
-    "module": "ES6",
+    "module": "es2022",
```

Comment:
Can we verify (manually is fine, doesn't need a test case) that attempting to use top-level await in some src/foo.ts file results in `npm run compile` *breaking* on the build of `build/src/`? That was the assertion made during the SIG discussion that decided to set the target as es2022 instead of es2020.